lib/acpi/acpi_shell.c

```diff
@@ -189,28 +196,26 @@ static int dump_dev_prs(const struct shell *sh, size_t argc, char **argv)
 static int dump_prt(const struct shell *sh, size_t argc, char **argv)
 {
 	IF_ENABLED(CONFIG_PCIE_PRT, ({
-		static ACPI_PCI_ROUTING_TABLE irq_prt_table[CONFIG_ACPI_MAX_PRT_ENTRY];
-		int status, cnt;
-		ACPI_PCI_ROUTING_TABLE *prt;
+		int irq, bus, dev_num, func;
+		pcie_bdf_t bdf;
 
-		if (argc < 2) {
-			shell_error(sh, "invalid arugment");
+		if (argc < 4) {
+			shell_error(sh, "invalid arguments [Eg: acpi prt <bus> <dev> <func>]");
 			return -EINVAL;
 		}
 
-		status = acpi_get_irq_routing_table(argv[1],
-						    irq_prt_table, ARRAY_SIZE(irq_prt_table));
-		if (status) {
-			return status;
-		}
-
-		prt = irq_prt_table;
-		for (cnt = 0; prt->Length; cnt++) {
-			shell_print(sh, "[%02X] PCI IRQ Routing Table Package", cnt);
-			shell_print(sh, "\tDevNum: %lld Pin: %d IRQ: %d",
-				    (prt->Address >> 16) & 0xFFFF, prt->Pin, prt->SourceIndex);
-
-			prt = ACPI_ADD_PTR(ACPI_PCI_ROUTING_TABLE, prt, prt->Length);
+		bus = atoi(argv[1]);
+		dev_num = atoi(argv[2]);
+		func = atoi(argv[3]);
+
+		bdf = PCIE_BDF(bus, dev_num, func);
+		irq = acpi_legacy_irq_get(bdf);
+		if (irq != UINT_MAX) {
+			shell_print(sh, "PCI Legacy IRQ number for bus %d device %d function %d is: %d",
+				 bus, dev_num, func, irq);
+		} else {
+			shell_print(sh, "PCI Legacy IRQ number for bus %d device %d function %d Not found",
```

Comment:
Compliance is complaining about the too long line. You could shorten it a bit by using "dev" and "func" instead of the full words